### DISCUSSION TITLE
How to Refresh Images Automatically After File Updates

### DISCUSSION THREAD

**Discussion by Mett-Barr** [coil-kt/coil#2747] [Q&A]
I'm using file paths as models with Coil on a PC, and I encountered a situation involving file copies and deletions. For example, if I have an image img, and I generate copies like img~1, img~2, etc., when I delete img~2 and then create a new copy, it will reuse the same name img~2 (according to the current naming rules).

The issue is that two different images, one deleted and one newly created, share the same file name. This causes Coil to not update the image since the path remains the same.

Is there a standard way to force Coil to refresh the image in such cases? I can listen to file changes—can this be used as a manual trigger to refresh the ImageLoader?

**Comment by colinrtwhite** [comment]
You could manually trigger the update with `AsyncImagePainter.restart`. Alternatively you could set `ImageLoader.Builder.addLastModifiedToFileCacheKey(true)` to include the last modified timestamp in the memory cache key.
